go/cmd/dolt/commands/diff.go

```diff
@@ -690,4 +766,6 @@ func writeDiffResults(
 			}
 		}
 	}
+
+	return writer.Close(ctx)
```

Comment:
(minor) might be just a little bit cleaner to have the `writer` closed in the function that creates it, instead of that being a side effect of `writeFilteredResults`. Still correct either way, but makes the lifetime of the `writer` resource a little easier to understand. I'd just use `defer writer.Close(ctx)` in the main function code that's calling this method.